### ISSUE TITLE
Safari:  Unexpected identifier "undefined"

### ISSUE THREAD

**Issue by DashBarkHuss** [WishTender/frontend#38]
**Environment**
Browser: Safari
Frontend: local
Backend: remote
Frontend commands: `npm start` and `npm run proxytoremote`

`npm start` and `npm run proxytoremote` error:
>Unhandled Rejection (SyntaxError): JSON Parse error: Unexpected identifier "undefined"
(anonymous function)
src/App.js:52

Recreate
1. run `npm run proxytoremote`
2. go to homepage

`npm run startlocal` and `npm run startlocalnotproxy` don't give this error .

**Comment by DashBarkHuss** [comment]
now I'm finding `npm run startlocal` and `npm run startlocalnotproxy` DO also give this error. Only the staging environment doesn't.

**Comment by DashBarkHuss** [comment]
Now I quit safari and reopened it and getting the error again.

I'm on `npm run startlocal` and one weird thing I see is that, though I cleared the cookies on `localhost:3000` and `localhost:4000` the request is still being sent with a cookie.
>Request
GET /api/users/current HTTP/1.1
Cookie: connect.sid=s%3ARwjUVY1npEG05S7BXdJyg4mC31CXXPit.3vDeWuAm%2FWTRgM4uq49Famr7DhFuIYHflYMCzBTUxhU; locale=%7B%22locale%22%3A%22en-us%22%2C%22countryCode%22%3A%22US%22%2C%22languageCode%22%3A%22en%22%7D
Accept: */*
Accept-Encoding: gzip, deflate
Host: localhost:3000
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15
Accept-Language: en-us
Referer: http://localhost:3000/
Connection: keep-alive

I fixed this by clearing the cookies in **Preferences**->**Privacy**->**Manage Website Data** and removed data from localhost. But I don't know why I couldn't  delete them in the cookies tab in the developer panel.

I created a sample create-react-app and server to isolate the issue. It is working on safari. Now I'm trying to see if the problem is in the back or front. So I swapped the wishtender backend with this sample backend. and it's not still not working. Which makes me think the problem is in the front.

Sample create-react-app and server  Request:
>Summary
URL: http://localhost:3001/api/users/current
Status: 200 OK
Source: Network
Address: 127.0.0.1:3001
Initiator: 
App.js:6
>
>Request
GET /api/users/current HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Host: localhost:3001
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15
Accept-Language: en-us
Referer: http://localhost:3001/
Connection: keep-alive
>
>Response
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
Access-Control-Allow-Origin
Set-Cookie: locale=en-US; Max-Age=1618512; Path=/; Expires=Tue, 04 May 2021 12:15:58 GMT; SameSite=Strict
Set-Cookie: connect.sid=s%3AqzEtNDLfKqdZlqMAt32IO9GpPIOS6jXB.MZsdo1X6RnlvG67o2p%2B5Gcq3%2FB2eHvqE5ad2Zr7ROtw; Path=/; HttpOnly; SameSite=Strict
Access-Control-Allow-Methods: GET, HEAD, POST, PUT, DELETE, TRACE, OPTIONS, PATCH
Access-Control-Allow-Headers: X-Requested-With, Accept, Content-Type
Date: Thu, 15 Apr 2021 18:40:46 GMT
Access-Control-Allow-Credentials: true
Content-Length: 34
Connection: close
ETag: W/"22-nIi03smINU2nZyWgQBjZdAzwinc"
Vary: Accept-Encoding
x-powered-by: Express

Both the locale cookie and connect.sid(session) cookie are visiuble in the cookie panel, and `document.cookie` gives `"locale=en-US" `

Wishtender front end and sample backend request:
> Summary
URL: http://localhost:3000/api/users/current
Status: 200 OK
Source: Network
Address: 127.0.0.1:3000
Initiator: 
UserContext.js:6
>
>Request
GET /api/users/current HTTP/1.1
Cookie: __stripe_mid=b5ef233d-79b3-4f24-99d7-70936aa7fe99727434; __stripe_sid=4326df7b-ed36-48af-a595-b7ffb59f4dc3f00975
Accept: */*
Accept-Encoding: gzip, deflate
Host: localhost:3000
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15
Accept-Language: en-us
Referer: http://localhost:3000/
Connection: keep-alive
>
>Response
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
Access-Control-Allow-Origin
Set-Cookie: locale=en-US; Max-Age=1618512; Path=/; Expires=Tue, 04 May 2021 12:11:33 GMT; SameSite=Strict
Set-Cookie: connect.sid=s%3ASASXPxJT_E9h_3YLP-gRSnyvbaO-1Cue.ycYTYmo6mdCd%2ByoRnEpJiUSzNXDCrj9BaOhQZgX%2Ba6Y; Path=/; HttpOnly; SameSite=Strict
Access-Control-Allow-Methods: GET, HEAD, POST, PUT, DELETE, TRACE, OPTIONS, PATCH
Access-Control-Allow-Headers: X-Requested-With, Accept, Content-Type
Date: Thu, 15 Apr 2021 18:36:21 GMT
Access-Control-Allow-Credentials: true
Content-Length: 34
Connection: close
ETag: W/"22-nIi03smINU2nZyWgQBjZdAzwinc"
Vary: Accept-Encoding
x-powered-by: Express

Neighter the locale cookie or connect.sid(session) cookie are visible in the cookie panel, and `document.cookie` gives nothing. But when you send the request again both cookies are in the request. So they are storing somewhere but I don't see them on the developer tools cookie panel on localhost3000 or 4000. The cookie is gone from the request if I go to **Preferences**->**Privacy**->**Manage Website Data** and remove data from localhost. So it's storing on some localhost. But since it's visible on localhost3000 or 4000.

I changed the sample react app to use `http-proxy-middleware` like wishtender does but with the proxy hard coded at 4000, and now I'm still getting the cookies in the cookie panel and console.

**Comment by DashBarkHuss** [comment]
I tried making the two apps more and more similar by changing the `package.json` and the `App.js`, etc. Until I realized the only difference left is that one is running on `localhost:3000` and the other `3001.` So I ran wishtender on `3001` and it worked. Then I switched it back to `3000` and it worked again- the cookies were available in the cookie panel. So I don't know what was wrong or how this works, but next time I will just switch the port and see if it helps. `PORT=3002 npm run startlocal`
